test/integration/apiserver/openapi_test.go

```diff
@@ -0,0 +1,115 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package apiserver
+
+import (
+	"encoding/json"
+	"net/http"
+	"testing"
+
+	"k8s.io/apiserver/pkg/features"
+	utilfeature "k8s.io/apiserver/pkg/util/feature"
+	"k8s.io/apiserver/pkg/util/openapi"
+	restclient "k8s.io/client-go/rest"
+	featuregatetesting "k8s.io/component-base/featuregate/testing"
+	"k8s.io/kube-openapi/pkg/common"
+	"k8s.io/kube-openapi/pkg/validation/spec"
+	generated "k8s.io/kubernetes/pkg/generated/openapi"
+	"k8s.io/kubernetes/test/integration/framework"
+)
+
+func TestEnablingOpenAPIEnumTypes(t *testing.T) {
+	const typeToAddEnum = "k8s.io/api/core/v1.ContainerPort"
+	const typeToCheckEnum = "io.k8s.api.core.v1.ContainerPort"
+
+	for _, tc := range []struct {
+		name            string
+		featureEnabled  bool
+		enumShouldExist bool
+	}{
+		{
+			name:            "disabled",
+			featureEnabled:  false,
+			enumShouldExist: false,
+		},
+		{
+			name:            "enabled",
+			featureEnabled:  true,
+			enumShouldExist: true,
+		},
```

Comment:
That's an interesting table-driven test, not sure this is specifically useful in that form :-)

Reply:
Still better than manually set/unset feature gate though.

Reply:
(And I can use IDE to run only some case, during initial development.